src/portfolio/PortfolioItem.js

```diff
@@ -16,7 +16,8 @@ export default class PortfolioItem extends PureComponent {
 
     render() {
         const { contract } = this.props;
-
+        const backendFailure = contract.validation_error && Object.keys(contract).length < 3;
```

Comment:
Extract this logic into a backendFailure = contract => thing and reuse.
Add a single line of comment that this is hackish.
Ideally will remove when backend starts behaving.